example/server.js

```diff
@@ -2,22 +2,51 @@
 
 var Hapi = require('hapi');
 var Hoek = require('hoek');
-var tags = require('../lib/index.js');
+var fs = require('fs');
+var pg = require('pg');
 
 var server = new Hapi.Server();
+var tagsData = JSON.parse(fs.readFileSync('./example/tags.json').toString());
+var tags = require('../lib/index.js');
+var tagsPool = new pg.Pool();
+
+
+function init (port, callback) {
+  server.connection({ port: port });
+
+  server.register([{
+    register: tags,
+    options: { tags: tagsData, tagsPool: tagsPool }
+  }], function (err) {
+    if (err) {
+      return callback(err);
+    }
+
+    server.route([{
+      method: 'GET',
+      path: '/',
+      handler: function (request, reply) {
+        tagsPool.connect(function (connErr, client, done) {
+          Hoek.assert(!connErr, connErr);
+          client.query('select * from tags', function (dberr, res) {
+            Hoek.assert(!dberr, dberr);
+            done();
 
-server.connection({ port: process.env.PORT || 3000 });
+            return reply(res.rows);
+          });
+        });
+      }
+    }, {
+      method: 'GET',
+      path: '/hello',
+      handler: function (request, reply) {
+        reply('hello');
+      }
+    }]);
 
-server.register(tags, function (err) {
-  Hoek.assert(!err, 'error registering plugin');
-});
+    return callback(null, server);
+  });
+}
 
-server.route({
-  method: 'GET',
-  path: '/',
-  handler: function (request, reply) {
-    reply(request.jackmisawesome);
-  }
-});
 
-module.exports = server;
+module.exports = { init: init, tagsPool: tagsPool };
```

Comment:
Sure, just to be clear, I don't think the way I am exporting the pool is nice, I just thought it was a low priority to work out how to export it so kept it as simple as possible. 

I am not really sure that exposing it via the init function is the best way, but if you want me to change it now, just let me know and I can look into it.